spec/specification/json_spec.rb

```diff
@@ -445,6 +464,17 @@ module Pod
         ]
       end
 
+      it 'can load on demand resources from json' do
+        json = '{"on_demand_resources":{"tag1":{"paths":"Tag1Resources/*","category":"on_demand"},"tag2":"Tag2Resources/*","tag3":["Tag3Resources/*"]}}'
+        result = Specification.from_json(json)
+        result.on_demand_resources.count.should.equal 3
+        result.on_demand_resources.should == {
+          'tag1' => { 'paths' => 'Tag1Resources/*', 'category' => 'on_demand' },
```

Comment:
the specification just stores the attribute which means the key can be a string if its coming from JSON. Using `consumer` would normalize this and ensure all keys are symbols, so this is fine.

Reply:
What's the difference between `on_demand` and `download_on_demand`?

Reply:
good catch this was a bad test! I had started this PR calling it `on_demand` and then switched to `download_on_demand` to be explicit. will update!